pandas/core/internals/blocks.py

```diff
@@ -2105,6 +2107,14 @@ def to_native_types(
     **kwargs,
 ) -> np.ndarray:
     """convert to our native types format"""
+    if isinstance(values, Categorical):
+        # GH#40754 Convert categorical datetimes to datetime array
+        values = take_nd(
+            values.categories._values,
+            ensure_platform_int(values._codes),
+            fill_value=na_rep,
```

Comment:
if the user doesn't pass na_rep, i think we're going to end up with with "nan" where we likely want "NaT"?

Categorical._internal_get_values may be helpful here?

Reply:
No, if really no value is passed ``na_value_for_dtype(arr.dtype, compat=False)`` is called, so this should be fine?

Reply:
ill take your word for it.